src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AbstractRemoveEmptyFinalizers.cs

```diff
@@ -35,72 +33,26 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            analysisContext.RegisterCompilationStartAction(compilationContext =>
+            analysisContext.RegisterCodeBlockAction(methodContext =>
             {
-                INamedTypeSymbol conditionalAttributeSymbol = WellKnownTypes.ConditionalAttribute(compilationContext.Compilation);
-
-                compilationContext.RegisterOperationBlockActionInternal(context =>
-                {
-                    var method = context.OwningSymbol as IMethodSymbol;
-                    if (method == null)
-                    {
-                        return;
-                    }
-
-                    if (!method.IsFinalizer())
-                    {
-                        return;
-                    }
-
-                    if (IsEmptyFinalizer(context.OperationBlocks, conditionalAttributeSymbol))
-                    {
-                        context.ReportDiagnostic(context.OwningSymbol.CreateDiagnostic(Rule));
-                    }
-                });
-            });
-        }
-
-        private static bool IsEmptyFinalizer(ImmutableArray<IOperation> operationBlocks, ISymbol conditionalAttributeSymbol)
-        {
-            if (operationBlocks != null && operationBlocks.Length == 1)
-            {
-                var block = operationBlocks[0] as IBlockStatement;
-                if (block == null)
+                var methodSymbol = (IMethodSymbol)methodContext.OwningSymbol;
```

Comment:
This may not always be a method symbol. For example, this will also be called for a field initializer.